src/messages.rs

```diff
@@ -530,6 +530,23 @@ pub fn command_complete(command: &str) -> BytesMut {
     res
 }
 
+/// Create a notify message.
+pub fn notify(message: &str, details: String) -> BytesMut {
+    let mut notify_cmd = BytesMut::new();
+
+    notify_cmd.put_slice("SNOTICE\0".as_bytes());
+    notify_cmd.put_slice("C00000\0".as_bytes());
+    notify_cmd.put_slice(format!("M{}\0", message).as_bytes());
+    notify_cmd.put_slice(format!("D{}\0", details).as_bytes());
+
+    let mut res = BytesMut::new();
+    res.put_u8(b'N');
+    res.put_i32(notify_cmd.len() as i32 + 4);
```

Comment:
Subtract one (`-1`). The length of the message does not include the message identifier (`N`).

Reply:
when chaging to +3, like this:

```diff
diff --git a/src/messages.rs b/src/messages.rs
index cb3b595..b1cd3f9 100644
--- a/src/messages.rs
+++ b/src/messages.rs
@@ -541,7 +541,7 @@ pub fn notify(message: &str, details: String) -> BytesMut {
 
     let mut res = BytesMut::new();
     res.put_u8(b'N');
-    res.put_i32(notify_cmd.len() as i32 + 4);
+    res.put_i32(notify_cmd.len() as i32 + 3);
     res.put(notify_cmd);
 
     res

```

it drops the connection:

```

❯ PGPASSWORD=postgres psql -h 127.0.0.1 -p 6432 -U postgres pgbouncer -c 'show help';
NOTICE:  Console usage
DETAIL:
	SHOW HELP|CONFIG|DATABASES|POOLS|CLIENTS|SERVERS|USERS|VERSION
	SHOW LISTS
	SHOW STATS
	SET key = arg
	RELOAD
	PAUSE [<db>, <user>]
	RESUME [<db>, <user>]
	SHUTDOWN
message contents do not agree with length in message type "N"
lost synchronization with server: got message type "
connection to server was lost
```

Reply:
From the log:

```
[2023-07-13T20:43:36.328145Z WARN  pgcat] Client disconnected with error SocketError("Error reading message code from socket - Error Kind(UnexpectedEof)")

```

Reply:
Ah nevermind, it must be something else.

Reply:
do you have any idea how to troubleshoot this?

Reply:
Ahh! I think there is a missing NULL byte. This is how we format the error message (same format as notice): https://github.com/postgresml/pgcat/blob/main/src/messages.rs#L357-L358

Reply:
It worked! 💪 

I've rebased and added a better commit message.

Is there anything else that you need me to change?

Reply:
looks great!